soc/xtensa/intel_adsp/cavs/gdbstub.c

```diff
@@ -0,0 +1,1394 @@
+	},
+	{
+		/* AR35 */
+		.idx = 0x123,
+		.regno = 0x123,
+		.byte_size = 4,
+		.gpkt_offset = 144,
+	},
+	{
+		/* AR36 */
+		.idx = 0x124,
+		.regno = 0x124,
+		.byte_size = 4,
+		.gpkt_offset = 148,
+	},
+	{
+		/* AR37 */
+		.idx = 0x125,
+		.regno = 0x125,
+		.byte_size = 4,
+		.gpkt_offset = 152,
+	},
+	{
+		/* AR38 */
+		.idx = 0x126,
+		.regno = 0x126,
+		.byte_size = 4,
+		.gpkt_offset = 156,
+	},
+	{
+		/* AR39 */
+		.idx = 0x127,
+		.regno = 0x127,
+		.byte_size = 4,
+		.gpkt_offset = 160,
+	},
+	{
+		/* AR40 */
+		.idx = 0x128,
+		.regno = 0x128,
+		.byte_size = 4,
+		.gpkt_offset = 164,
+	},
+	{
+		/* AR41 */
+		.idx = 0x129,
+		.regno = 0x129,
+		.byte_size = 4,
+		.gpkt_offset = 168,
+	},
+	{
+		/* AR42 */
+		.idx = 0x12a,
+		.regno = 0x12a,
+		.byte_size = 4,
+		.gpkt_offset = 172,
+	},
+	{
+		/* AR43 */
+		.idx = 0x12b,
+		.regno = 0x12b,
+		.byte_size = 4,
+		.gpkt_offset = 176,
+	},
+	{
+		/* AR44 */
+		.idx = 0x12c,
+		.regno = 0x12c,
+		.byte_size = 4,
+		.gpkt_offset = 180,
+	},
+	{
+		/* AR45 */
+		.idx = 0x12d,
+		.regno = 0x12d,
+		.byte_size = 4,
+		.gpkt_offset = 184,
+	},
+	{
+		/* AR46 */
+		.idx = 0x12e,
+		.regno = 0x12e,
+		.byte_size = 4,
+		.gpkt_offset = 188,
+	},
+	{
+		/* AR47 */
+		.idx = 0x12f,
+		.regno = 0x12f,
+		.byte_size = 4,
+		.gpkt_offset = 192,
+	},
+	{
+		/* AR48 */
+		.idx = 0x130,
+		.regno = 0x130,
+		.byte_size = 4,
+		.gpkt_offset = 196,
+	},
+	{
+		/* AR49 */
+		.idx = 0x131,
+		.regno = 0x131,
+		.byte_size = 4,
+		.gpkt_offset = 200,
+	},
+	{
+		/* AR50 */
+		.idx = 0x132,
+		.regno = 0x132,
+		.byte_size = 4,
+		.gpkt_offset = 204,
+	},
+	{
+		/* AR51 */
+		.idx = 0x133,
+		.regno = 0x133,
+		.byte_size = 4,
+		.gpkt_offset = 208,
+	},
+	{
+		/* AR52 */
+		.idx = 0x134,
+		.regno = 0x134,
+		.byte_size = 4,
+		.gpkt_offset = 212,
+	},
+	{
+		/* AR53 */
+		.idx = 0x135,
+		.regno = 0x135,
+		.byte_size = 4,
+		.gpkt_offset = 216,
+	},
+	{
+		/* AR54 */
+		.idx = 0x136,
+		.regno = 0x136,
+		.byte_size = 4,
+		.gpkt_offset = 220,
+	},
+	{
+		/* AR55 */
+		.idx = 0x137,
+		.regno = 0x137,
+		.byte_size = 4,
+		.gpkt_offset = 224,
+	},
+	{
+		/* AR56 */
+		.idx = 0x138,
+		.regno = 0x138,
+		.byte_size = 4,
+		.gpkt_offset = 228,
+	},
+	{
+		/* AR57 */
+		.idx = 0x139,
+		.regno = 0x139,
+		.byte_size = 4,
+		.gpkt_offset = 232,
+	},
+	{
+		/* AR58 */
+		.idx = 0x13a,
+		.regno = 0x13a,
+		.byte_size = 4,
+		.gpkt_offset = 236,
+	},
+	{
+		/* AR59 */
+		.idx = 0x13b,
+		.regno = 0x13b,
+		.byte_size = 4,
+		.gpkt_offset = 240,
+	},
+	{
+		/* AR60 */
+		.idx = 0x13c,
+		.regno = 0x13c,
+		.byte_size = 4,
+		.gpkt_offset = 244,
+	},
+	{
+		/* AR61 */
+		.idx = 0x13d,
+		.regno = 0x13d,
+		.byte_size = 4,
+		.gpkt_offset = 248,
+	},
+	{
+		/* AR62 */
+		.idx = 0x13e,
+		.regno = 0x13e,
+		.byte_size = 4,
+		.gpkt_offset = 252,
+	},
+	{
+		/* AR63 */
+		.idx = 0x13f,
+		.regno = 0x13f,
+		.byte_size = 4,
+		.gpkt_offset = 256,
+	},
+	{
+		/* LBEG */
+		.idx = 512,
+		.regno = 0x0200,
```

Comment:
Style: seems like idx == regno always?  If so, would be nice to enforce that.  Also would be nice to have them written in the same base so the relationship is obvious to the reader.

Reply:
While they are the same for xt-gdb, they differ for the open source toolchain.